content/blog/HTML Select Uses/Index.md

```diff
@@ -0,0 +1,90 @@
+---
+{
+    title: "HTML Select Uses",
+    description: "An introduction to HTML's input:list and select boxes.",
+    published: '2024-03-06',
+    authors: ['jerico'],
+    tags: ['select', 'javascript'],
+    license: 'cc-by-4',
+    originalLink: ""
+}
+---
+
+In this short article, I will introduce you to HTML’s Input:List and DataList elements/attributes. 
+
+I will also cover their functionality, introduce bits of JavaScript, and then show how to expand on this.
+
+##  Elements And Order
+
+`List` is an input attribute that allows a person to select boxes and values for a user to select themselves. You specify where the list is getting its data or options from bypassing the `Id` of a `dataList` element into it. Then in the dataList element, you put your `option` element with the attribute of `value` to give the value of the select box or input (the return).
+
+Here is a starting example using `Input`, `datalist`, and `option` elements.
+
+```
+<input list=”optionsList”>
+
+<datalist id=”optionsList”>
+	<option value=”Thing1”> <!--Option element does not need closing tag (</option>)-->
+	<option value=”Thing2”>
+	<option value=”Thing3”>
+</datalist>
+```
+
+In the example above I give `Thing#` as one of the options. If you were a user trying to select one of these options it would look something like this.
+
+ ![Options, input, datalist example.](./optionsPic.png)
+
+## Reasonable Use 
+
+Why would I use this in my own work? The reason is, that it has functionality and allows you to only get back the values you want. For example, If you wanted to have a list of Zipcodes that a user could select from instead of having them type their Zipcode.
+
+Note: A user still can type in this `input:list` box but all that does is search all the options for you and give you the ones that you can select.
+
+## Simple Application
+
+In the note before this section, I mentioned how “A user still can type in this `input:list` box but all that does is search all the options for you and give you the remaining ones that you can select”.  With that in mind, I want to show how you can return the remaining options in the input with `JavaScript`. 
+
+The code will have comments, so there is no further explanation. The example below is an expansion on the ZipCodes example.
+JsFiddle for quick viewing https://jsfiddle.net/dek12pgj/. 
```

Comment:
We nowadays prefer to avoid having `jsfiddle` links and such. Instead, we have a way of embedding an iframe of a code sample with the code kept in our Git folder.

That way, if this link ever changes, it won't break this article. Let's talk more in our Discord to figure this out :) 

https://discord.gg/FMcvc6T

Reply:
ok, give me a few minutes and I should be there.

Reply:
@crutchcorn would there be an easier way to discuss this instead of discord? Like email?